src/Confluent.Kafka/Serialization/StringSerializer.cs

```diff
@@ -37,21 +44,71 @@ public StringSerializer(Encoding encoding)
         }
 
         /// <summary>
+        ///     Initializes a new StringSerializer class instance.
+        ///     The encoding encoding to use must be provided via <see cref="Producer" /> configuration properties.
+        /// </summary>
+        public StringSerializer()
+        {
+        }
+
+        /// <summary>
         ///     Encodes a string value in a byte array.
         /// </summary>
-        /// <param name="val">
+        /// <param name="data">
         ///     The string value to serialize.
         /// </param>
+        /// <param name="topic">
+        ///     The topic associated with the data (ignored by this serializer).
+        /// </param>
         /// <returns>
-        ///     <paramref name="val" /> encoded in a byte array (or null if <paramref name="val" /> is null).
+        ///     <paramref name="data" /> encoded in a byte array (or null if <paramref name="data" /> is null).
         /// </returns>
-        public byte[] Serialize(string val)
+        public byte[] Serialize(string topic, string data)
         {
-            if (val == null)
+            if (data == null)
             {
                 return null;
             }
-            return encoding.GetBytes(val);
+
+            return encoding.GetBytes(data);
         }
+
+        /// <include file='../include_docs.xml' path='API/Member[@name="ISerializer_Configure"]/*' />
+        public IEnumerable<KeyValuePair<string, object>> Configure(IEnumerable<KeyValuePair<string, object>> config, bool isKey)
+        {
+            var propertyName = isKey ? KeyEncodingConfigParam : ValueEncodingConfigParam;
+            var keyOrValue = isKey ? "Key" : "Value";
+
+            if (config.Count(ci => ci.Key == propertyName) > 0)
+            {
+                try
+                {
+                    encoding = config
+                        .Single(ci => ci.Key == propertyName)
+                        .Value
+                        .ToString()
+                        .ToEncoding();
+                }
+                catch
+                {
+                    throw new ArgumentException($"{keyOrValue} StringSerializer encoding configuration parameter was specified twice.");
```

Comment:
ToEncoding may also throw, separate both (first "Signle" with the try/catch with this message, second with ToEncoding witout try catch as it will simply throw if necessary
Also, when you throw an exception from a catch block, you should include the exception catch in InnerException so you keep the origin (here we would have lost the encoding issue for example)